### PR TITLE
Cuibackofficepages

### PR DESCRIPTION
**Pull request by rajguptaH**

### CODE REVIEW

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/UiPageTypeRepository.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            using IDbConnection db = _connectionFactory.GetConnection;
+            return db.Query<UiPageTypeModel>("Select * From [UiPageType] where IsDeleted=0").ToList();
+        }
+        /// <summary>
+        /// Getting All Record Data Type 
+        /// </summary>
+        /// <returns></returns>
+
+        public List<DataTypeModel> GetDataType()
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/UiPageTypeRepository.cs:60]: why is this method here?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IUiPageTypeRepository.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+namespace TestingAndCalibrationLabs.Business.Data.Repository.Interfaces
+{
+    /// <summary>
+    /// Repository interface for Ui Page Type
+    /// </summary>
+    public interface IUiPageTypeRepository
+    {
+        List<UiPageTypeModel> GetAll();
+        List<DataTypeModel> GetDataType();
+        List<UiPageValidationTypeModel> GetUiPageValidationType();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IUiPageTypeRepository.cs:15]: why is this methods here?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiControlTypeController.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        /// <returns></returns>
+        [HttpGet]
+        public IActionResult Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+            Business.Core.Model.UiControlTypeModel conModel = _uiControlTypeServices.Get((int)id);
+            if (conModel == null)
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiControlTypeController.cs:49]: conModel?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        }
+        /// <summary>
+        /// To List All Record
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public IActionResult Index()
+        {
+            ViewBag.IsSuccess = TempData["IsTrue"] != null ? TempData["IsTrue"] : false;
+            List<Business.Core.Model.UiPageMetadataModel>pageCon = _uiPageMetadataService.GetAll();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:39]: pageCon?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiControlTypeController.cs (hunk 3 above)
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiControlTypeController.cs:49]: ya i asked the same question can i use object names as well as or change but you replied me with - Object name as well. this

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IUiPageTypeRepository.cs (hunk 2 above)
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IUiPageTypeRepository.cs:15]: this is for getting list of datatypes for dropdowns

Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IUiPageTypeRepository.cs:15]: and uipagevalidation type

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        /// <summary>
+        /// For Create Record View
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet]
+        public IActionResult Create(int id)
+        {
+            
+            List<Business.Core.Model.UiPageTypeModel> page = _uiPageTypeService.GetAll();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:53]: rename it to pages

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            List < Business.Core.Model.UiControlTypeModel>control = _uiControlTypeService.GetAll();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:54]: name it controls

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/UiPageTypeRepository.cs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            using IDbConnection db = _connectionFactory.GetConnection;
+            return db.Query<DataTypeModel>("SELECT d.Name as Name, d.Id as Id from DataType d where IsDeleted=0").ToList();
+      }
+        
+        /// <summary>
+        /// Getting All Record Ui Page Validation Type 
+        /// </summary>
+        /// <returns></returns>
+
+        public List<UiPageValidationTypeModel> GetUiPageValidationType()
```
[the remark this answers is not in the document]
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/UiPageTypeRepository.cs:71]: i used those tables getList method in UiPageTypeRepository becuase there is no already made methods for these tables

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+            List<Business.Core.Model.DataTypeModel> data = _uiPageTypeService.GetDataType();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:55]: name it dataTypes

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+            var pagess = _mapper.Map<List<Business.Core.Model.UiPageTypeModel>, List<Models.UiPageTypeModel>>(page);
+            var controless = _mapper.Map<List<Business.Core.Model.UiControlTypeModel>, List<Models.UiControlTypeModel>>(control);
+            var datass = _mapper.Map<List<Business.Core.Model.DataTypeModel>, List<Models.UiDataTypeModel>>(data);
+            ViewBag.Control = controless.ToList();
+            ViewBag.Data= datass.ToList();
+            ViewBag.Page = pagess.ToList();
+
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:62]: What is thsi variable naem controless?
name the viewbag like Control to ViewBag.UiControlTypes and so on for all other

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+            List<Business.Core.Model.UiControlTypeModel> control = _uiControlTypeService.GetAll();
+            List<Business.Core.Model.DataTypeModel> data = _uiPageTypeService.GetDataType();
+            var pageList = _mapper.Map<List<Business.Core.Model.UiPageTypeModel>, List<Models.UiPageTypeModel>>(page);
+            var controlList = _mapper.Map<List<Business.Core.Model.UiControlTypeModel>, List<Models.UiControlTypeModel>>(control);
+            var dataList = _mapper.Map<List<Business.Core.Model.DataTypeModel>, List<Models.UiDataTypeModel>>(data);
+            ViewBag.Control = controlList.ToList();
+            ViewBag.Data = dataList.ToList();
+            ViewBag.page = pageList.ToList();
+            Business.Core.Model.UiPageMetadataModel pMeta = _uiPageMetadataService.GetById((int)id);
+            var pageCons = _mapper.Map<Business.Core.Model.UiPageMetadataModel, Models.UiPageMetadataDTO>(pMeta);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:111]: all the variable names and View bag should be named like above given comments

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, [Bind] Models.UiPageMetadataDTO pageModel)
+        {
+            if (id != pageModel.Id)
+            {
+                return NotFound();
+            }
+            if (ModelState.IsValid)
+            {
+                var editMeta = _mapper.Map<Models.UiPageMetadataDTO, Business.Core.Model.UiPageMetadataModel>(pageModel);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:130]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public IActionResult Delete(int id)
+        {
+            if(id == null)
+            {
+                return NotFound();
+            }
+            Business.Core.Model.UiPageMetadataModel pcon = _uiPageMetadataService.GetById((int)id);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:147]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+            var deleteM = _mapper.Map<Business.Core.Model.UiPageMetadataModel, Models.UiPageMetadataDTO>(pcon);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageMetadataController.cs:148]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet]
+        public IActionResult Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+            var pageModel = _uiPageTypeService.GetById((int)id);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs:49]: meaningful variable name

Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs:49]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, [Bind] Models.UiPageTypeModel pageModel)
+        {
+            if (id != pageModel.Id)
+            {
+                return NotFound();
+            }
+            if (ModelState.IsValid)
+            {
+                var pageData = _mapper.Map<Models.UiPageTypeModel, Business.Core.Model.UiPageTypeModel>(pageModel);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs:74]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+        /// <param name="pageModel"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Create([Bind] Models.UiPageTypeModel pageModel)
+        {
+            if (ModelState.IsValid)
+            {
+               
+                var pageCreate = _mapper.Map<Models.UiPageTypeModel, Business.Core.Model.UiPageTypeModel>(pageModel);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs:102]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+            if (id == null)
+            {
+                return NotFound();
+            }
+            Business.Core.Model.UiPageTypeModel con = _uiPageTypeService.GetById((int)id);
+            if (con == null)
+            {
+                return NotFound();
+            }
+            var pageDel = _mapper.Map<Business.Core.Model.UiPageTypeModel, Models.UiPageTypeModel>(con);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageTypeController.cs:125]: meaningful variable name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageValidationController.cs [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+        /// <summary>
+        /// Get All The Pages From Database
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public IActionResult Index()
+        {
+            ViewBag.IsSuccess = TempData["IsTrue"] != null ? TempData["IsTrue"] : false;
+            List<Business.Core.Model.UiPageValidationModel> pageReq = _uiPageValidationTypeService.GetAll();
+            var pvList = _mapper.Map<List<Business.Core.Model.UiPageValidationModel>, List<Models.UiPageValidationModel>>(pageReq);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/UiPageValidationController.cs:38]: meaningful variable name

**Review by smartdusttechnologies — COMMENTED** [review]
is this file is required to be pushed?
TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/TestingAndCalibrationLabs.Web.UI.csproj.user

**Pull request closed without merging**